### ISSUE TITLE
Created Resources by Application are deleted when user make a change, occuring with no pattern

### ISSUE THREAD

**Issue by gokcesurenkok** [kubevela/kubevela#6221]
**Describe the bug**
Hello , We discovered this when we added some annotations to one of our applications which holds kustomize and kustomize patch components within, when we added new fields to our kustomize patch component, we observed that the applications itself stayed however created kustomization and gitrepository fluxcd crds were deleted , and stayed that way for a while, after some moment they are recreated again. 

And we see that resource tracker object of this application is also gone.

First, we thought there was something wrong with kustomization patch, we did many trials but there was no hope, then we suspected from the upgrade of vela controller , we were using 1.8.1 and upgraded to 1.9.3, and we observed that previously created applications having vela 1.8.0 labels on applications, so we deleted the applications and observed that newly created applications got the 1.9.3 label.

however, the problem was still there, most recently we also discovered that controller-sharded vela-core-controller replicas are running 1.8.0 image however main deployment has 1.9.3. it is because vela sharding controller addon patches the main deployment to create sub-deployments, so I deleted sub-deployments and re-enabled the addon to create correct deployments running 1.9.3. 

Currently, we are still testing, because it doesn't happen always, it seems to be happening randomly and rarely

**To Reproduce**

1. add some annotations to your application yaml via gitops 
2. observe the created sub resources
3. see they are all deleted along with application revision tracker object too

**Screenshots**

**KubeVela Version**

CLI Version: 1.9.3
Core Version: 1.9.3
GitRevision: 4637e3a918c51db95cd1caf3586bd91687bfe654
GolangVersion: go1.20.5

**Cluster information**

- Kubernetes Version - v1.23.7

**Additional context**

observers: @previousdeveloper @erkanzileli

**Comment by previousdeveloper** [comment]
Thanks for your help; it can be beneficial for our customers. Do you know the main reason for the case? Why does version annotation solve resource deletion?

**Comment by wonderflow** [comment]
Yes, using [publishVersion](https://kubevela.io/docs/end-user/version-control/#introduction) annotation on Application CR, so that after application created with a specified publish version, no changes of definition or application spec will affect the running Instance unless you change the publishVersion again.

If you don't set this version, everytime you change the application CR, it will trigger a deploy new version action which will apply all resources rendered out from Component and Trait definitions and apply to Kubernetes system. If the underlying resources' spec changes (such as the kustomization and gitrepository resources from fluxcd you mentioned), then it will trigger the update. Update annations equals changes the spec.

**Comment by skyshaokai** [comment]
> Thanks for your help; it can be beneficial for our customers. Do you know the main reason for the case? Why does version annotation solve resource deletion? @skyshaokai

Kubevela logs show that when changes are made, if publishVersion is not configured, there will be logs that delete the current version configuration. The reason seems to be that Kubevela has a logic for calculating the configuration content version. In addition, there is a configuration version of GC that recycles configuration content logic. This is achieved in two places and there is a logical inconsistency. There was an error during the GC process, thinking that it was necessary to GC the configuration content of the current version. Causes the resource to be deleted. When the publishVersion is configured, it should be possible to avoid this inconsistency. When using GC, the publishVersion should also prevail. Only when the publishVersion changes will historical versions of resources be deleted.

The above content is for reference only, and I did not check the source code to confirm this issue.

**Comment by anoop2811** [comment]
ping @chivalryq @dhiguero

**Comment by chivalryq** [comment]
I am still monitoring this situation closely, but unfortunately, there is no reliable way to replicate the results. :(

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
